hudi-common/src/test/java/org/apache/hudi/common/testutils/FileCreateUtils.java

```diff
@@ -219,13 +221,19 @@ public static void createBaseFile(String basePath, String partitionPath, String
 
   public static void createBaseFile(String basePath, String partitionPath, String instantTime, String fileId, long length)
       throws Exception {
+    createBaseFile(basePath, partitionPath, instantTime, fileId, length, Instant.now().toEpochMilli());
+  }
+
+  public static void createBaseFile(String basePath, String partitionPath, String instantTime, String fileId, long length, long lastModificationTimeMilli)
+      throws Exception {
     Path parentPath = Paths.get(basePath, partitionPath);
     Files.createDirectories(parentPath);
     Path baseFilePath = parentPath.resolve(baseFileName(instantTime, fileId));
     if (Files.notExists(baseFilePath)) {
       Files.createFile(baseFilePath);
     }
     new RandomAccessFile(baseFilePath.toFile(), "rw").setLength(length);
+    Files.setLastModifiedTime(baseFilePath, FileTime.fromMillis(lastModificationTimeMilli));
```

Comment:
may I know why we need to set mod time explicitly. I thought these are newly created files and so modificaiton time will be set to creation time and these are immutable files. can you help me understand.

Reply:
@nsivabalan the problem comes from mod time being the same for multiple input files. I uploaded the screenshot in the JIRA ticket. Also posting here for easy illustration

![Screen Shot 2021-03-26 at 1 42 42 AM](https://user-images.githubusercontent.com/2701446/119078833-cde7d180-b9ab-11eb-9d0f-32625dc30b3c.png)

DFSPathSelector reads last modification time and saves it as checkpoint, which is then used to compare with next batch of input files. It's not about files being mutated; the input files are append-only and last mod time _is_ create time. The test setup is to ensure last mod time being the same to avoid code execution causing delays when creating them.